zetaclient/chains/evm/observer/v2_inbound.go

```diff
@@ -149,7 +151,58 @@ func (ob *Observer) parseAndValidateDepositEvents(
 		return validEvents[i].Raw.BlockNumber < validEvents[j].Raw.BlockNumber
 	})
 
-	return validEvents
+	// Check if multiple calls are enabled
+	if ob.shouldAllowMultipleCalls(ctx) {
+		return validEvents
+	}
+	return ob.filterDepositEventsByTx(validEvents)
+}
+
+// shouldAllowMultipleCalls checks the EnableMultipleCalls feature flag
+func (ob *Observer) shouldAllowMultipleCalls(ctx context.Context) bool {
```

Comment:
actually i think this should be in `context` package, and we can just call `ztcx.IsEnableMultipleCallsEnabled(ctx)` because seems more generic and easier to use across codebase, but please let me know your thoughts before i refactor